chunking_test.py

```diff
@@ -0,0 +1,24 @@
+import json
+import os
+
+import zarr
+
+store = zarr.DirectoryStore("data/chunking_test.zarr")
+z = zarr.zeros((20, 3), chunks=(3, 3), shards=(2, 2), store=store, overwrite=True, compressor=None)
```

Comment:
Pondering the feedback on #877, I wonder if rather than the relatively hard-coded `shared=...`, an implementation of some interface here might not be able to give us the flexibility for different backends. i.e. basically passing in `chunk_store` directly, but that would raise the question of how to serialize the state into `.zarray` for other processes.